### ISSUE TITLE
JavadocTypeCheck incorrectly matches record component @param tags using prefix instead of exact match

### ISSUE THREAD

**Issue by Sumitkapoor129** [checkstyle/checkstyle#19770]
I noticed that the Javadoc check does not correctly validate `@param` tags for record components.

Currently, it treats a tag as valid if it shares a prefix with the component name, instead of requiring an exact match. This leads to inconsistent validation results.

---

### **Steps to reproduce**

1. Create a file with the following content:

```java
/**
 * Example case
 *
 * @param john123 wrong tag for a different component
 */
record Sample(String john) {}
```

2. Use the following Checkstyle configuration:

```xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType"/>
    </module>
</module>
```

3. Run Checkstyle:

```bash
checkstyle -c checkstyle.xml Sample.java
```

---

### **Expected behavior**

- `@param` tags should match the record component name **exactly**
- A missing `@param john` violation should be reported

---

### **Actual behavior**

- `@param john123` is treated as documenting `john` due to prefix-based matching  
- Missing `@param john` is **not reported**  
- `john123` is still flagged as an unused tag  

---

### **Why this is a problem**

This creates inconsistent behavior:
- The incorrect tag is accepted in one place  
- But rejected in another  

This makes the validation confusing and unreliable.

---

**Comment by romani** [comment]
Please strictly follow issue template , it will give us all details we need to help you.

**Comment by Sumitkapoor129** [comment]
> Please strictly follow issue template , it will give us all details we need to help you.

Changed it according to the template..

**Comment by romani** [comment]
>Run Checkstyle

We need `java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java`

Please share

**Comment by romani** [comment]
@mgustimz, please help us here to report issue in format we demand. 
I need to double check all to confirm that issue is valid.

**Comment by mgustimz** [comment]
Hi @romani 

I realized I submitted #19778 directly from my main branch instead of a separate one! Closing this out to keep things tidy.

I've opened a new PR with the exact same changes here: #19797 


Also this is the result by run checkstyle with old condition (before fixing) with the same step to reproduce above.
<img width="1657" height="103" alt="Image" src="https://github.com/user-attachments/assets/c526af89-a4bd-436b-888c-9e2bea5d5f23" />

**Comment by romani** [comment]
@mgustimz, please help us here to report issue in format we demand.
I need to double check all to confirm that issue is valid.


https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by romani** [comment]
this is how issue description should looks like https://github.com/checkstyle/checkstyle/issues/19717

**Comment by mgustimz** [comment]
Hi @romani 

I have read check documentation: https://checkstyle.org/checks/javadoc/javadocype.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

➜  Checkstyle javac Sample.java
➜  Checkstyle cat checkstyle.xml
```xml
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType"/>
  </module>
</module>%
```
➜  Checkstyle cat Sample.java
```java
/**
 * Example case
 *
 * @param john123 wrong tag for a different component
 */
record Sample(String john) {}%
```
➜  Checkstyle RUN_LOCALE="-Duser.language=en -Duser.country=US"
➜  Checkstyle java $RUN_LOCALE -jar checkstyle-13.4.2-all.jar -c checkstyle.xml Sample.java
Starting audit...
[ERROR] /Users/mgusti.maulanaz/Documents/Checkstyle/Sample.java:4:4: Unused @param tag for 'john123'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.

---

When a record has a component named `john` and its Javadoc contains `@param john123` (a tag for a different component, not `john`), `JavadocTypeCheck` should NOT report a missing `@param john` tag. The `@param john123` tag should only be flagged as an unused tag for a different component.

`JavadocTypeCheck` incorrectly matches `@param john123` to the record component `john` using prefix matching (`indexOf("john") == 0`), causing it to:
1. Report a false positive "missing @param john tag" error
2. Fail to correctly identify `@param john123` as an unused tag for a different component

**Expected behavior:** `@param` tags should match record components by exact name (or name + space for additional description), not by prefix. For example, `@param john123` should NOT match record component `john` because they are different parameter names.

---

**Comment by romani** [comment]
Please share it as text , not as image :)

**Comment by mgustimz** [comment]
@romani Sorry, now edited. Please check.
